File: app/src/main/java/com/muaj/oops/widgets/NoScrollViewPager.java
package com.muaj.oops.widgets;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

/** ViewPager
 *  user-defined whether horizontal slide
 * @author muaj
 * @date 2018/5/16
 */

public class NoScrollViewPager extends ViewPager {
    private boolean  noScroll=true;
    public NoScrollViewPager(@NonNull Context context) {
        super(context);
    }

    public NoScrollViewPager(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return !noScroll && super.onTouchEvent(ev);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return !noScroll && super.onInterceptTouchEvent(ev);
    }
}

File: lib/src/main/java/com/muaj/lib/view/NoScrollViewPager.kt
package com.muaj.lib.view

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

/**
 * Created by muaj on 2018/6/6
 * ViewPager
 * user-defined whether horizontal slide
 */

class NoScrollViewPager : ViewPager {
    private val noScroll = true

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return !noScroll && super.onTouchEvent(ev)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return !noScroll && super.onInterceptTouchEvent(ev)
    }
}
